images/virtualization-artifact/pkg/controller/cvi/internal/source/object_ref_vd.go

```diff
@@ -0,0 +1,247 @@
+
+		log.Info("Create importer pod...", "progress", cvi.Status.Progress, "pod.phase", "nil")
+
+		return requeue, nil
+	case cc.IsPodComplete(pod):
+		err = ds.statService.CheckPod(pod)
+		if err != nil {
+			cvi.Status.Phase = virtv2.ImageFailed
+
+			switch {
+			case errors.Is(err, service.ErrProvisioningFailed):
+				condition.Status = metav1.ConditionFalse
+				condition.Reason = vicondition.ProvisioningFailed
+				condition.Message = service.CapitalizeFirstLetter(err.Error() + ".")
+				return false, nil
+			default:
+				return false, err
+			}
+		}
+
+		condition.Status = metav1.ConditionTrue
+		condition.Reason = vicondition.Ready
+		condition.Message = ""
+
+		cvi.Status.Phase = virtv2.ImageReady
+		cvi.Status.Size = ds.statService.GetSize(pod)
+		cvi.Status.CDROM = ds.statService.GetCDROM(pod)
+		cvi.Status.Format = ds.statService.GetFormat(pod)
+		cvi.Status.Progress = "100%"
+		cvi.Status.Target.RegistryURL = ds.statService.GetDVCRImageName(pod)
+
+		log.Info("Ready", "progress", cvi.Status.Progress, "pod.phase", pod.Status.Phase)
+	default:
+		err = ds.statService.CheckPod(pod)
+		if err != nil {
+			cvi.Status.Phase = virtv2.ImageFailed
+
+			switch {
+			case errors.Is(err, service.ErrNotInitialized), errors.Is(err, service.ErrNotScheduled):
+				condition.Status = metav1.ConditionFalse
+				condition.Reason = vicondition.ProvisioningNotStarted
+				condition.Message = service.CapitalizeFirstLetter(err.Error() + ".")
+				return false, nil
+			case errors.Is(err, service.ErrProvisioningFailed):
+				condition.Status = metav1.ConditionFalse
+				condition.Reason = vicondition.ProvisioningFailed
+				condition.Message = service.CapitalizeFirstLetter(err.Error() + ".")
+				return false, nil
+			default:
+				return false, err
+			}
+		}
+
+		err = ds.importerService.Protect(ctx, pod)
+		if err != nil {
+			return false, err
+		}
+
+		condition.Status = metav1.ConditionFalse
+		condition.Reason = vicondition.Provisioning
+		condition.Message = "Import is in the process of provisioning to DVCR."
+
+		cvi.Status.Phase = virtv2.ImageProvisioning
+		cvi.Status.Progress = ds.statService.GetProgress(cvi.GetUID(), pod, cvi.Status.Progress)
+		cvi.Status.Target.RegistryURL = ds.statService.GetDVCRImageName(pod)
+
+		log.Info("Provisioning...", "progress", cvi.Status.Progress, "pod.phase", pod.Status.Phase)
+	}
+
+	return true, nil
+}
+
+func (ds ObjectRefVirtualDisk) CleanUpSupplements(ctx context.Context, cvi *virtv2.ClusterVirtualImage) (bool, error) {
+	supgen := supplements.NewGenerator(cc.CVIShortName, cvi.Name, ds.controllerNamespace, cvi.UID)
+
+	importerRequeue, err := ds.importerService.CleanUpSupplements(ctx, supgen)
+	if err != nil {
+		return false, err
+	}
+
+	diskRequeue, err := ds.diskService.CleanUpSupplements(ctx, supgen)
+	if err != nil {
+		return false, err
+	}
+
+	return importerRequeue || diskRequeue, nil
+}
+
+func (ds ObjectRefVirtualDisk) CleanUp(ctx context.Context, cvi *virtv2.ClusterVirtualImage) (bool, error) {
```

Comment:
CleanUp isn't called for `ok, err := source.CleanUp(ctx, cvi)`, but it should be called

Reply:
Fixed.